### ISSUE TITLE
Crashes with "0 bytes read""

### ISSUE THREAD

**Issue by lrvick** [mopidy/mopidy-gmusic#82]
When I enable gmusic it crashes as follows:

```
May 01 04:30:31 shell mopidy[2722]: ERROR    Unhandled exception in GMusicBackend (urn:uuid:2f6b6a7a-32d9-450a-bcbf-e8bd4a18f80b):
May 01 04:30:31 shell mopidy[2722]: Traceback (most recent call last):
May 01 04:30:31 shell mopidy[2722]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 191, in _actor_loop
May 01 04:30:31 shell mopidy[2722]: self.on_start()
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy_gmusic/actor.py", line 48, in on_start
May 01 04:30:31 shell mopidy[2722]: self.config['gmusic']['deviceid'])
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy_gmusic/session.py", line 21, in login
May 01 04:30:31 shell mopidy[2722]: if not self.api.login(username, password):
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/clients/mobileclient.py", line 39, in login
May 01 04:30:31 shell mopidy[2722]: if not self.session.login(email, password):
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 147, in login
May 01 04:30:31 shell mopidy[2722]: success = super(Mobileclient, self).login(email, password, *args, **kwargs)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 122, in login
May 01 04:30:31 shell mopidy[2722]: webclient.Init.perform(self, True)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/protocol/shared.py", line 208, in perform
May 01 04:30:31 shell mopidy[2722]: response = session.send(req_kwargs, cls.required_auth)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 86, in send
May 01 04:30:31 shell mopidy[2722]: res = self._send_with_auth(req_kwargs, desired_auth, rsession)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 142, in _send_with_auth
May 01 04:30:31 shell mopidy[2722]: return rsession.request(**req_kwargs)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 465, in request
May 01 04:30:31 shell mopidy[2722]: resp = self.send(prep, **send_kwargs)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 605, in send
May 01 04:30:31 shell mopidy[2722]: r.content
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 750, in content
May 01 04:30:31 shell mopidy[2722]: self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes()
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/models.py", line 673, in generate
May 01 04:30:31 shell mopidy[2722]: for chunk in self.raw.stream(chunk_size, decode_content=True):
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/response.py", line 304, in stream
May 01 04:30:31 shell mopidy[2722]: for line in self.read_chunked(amt):
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/response.py", line 411, in read_chunked
May 01 04:30:31 shell mopidy[2722]: raise httplib.IncompleteRead(''.join(line))
May 01 04:30:31 shell mopidy[2722]: IncompleteRead: IncompleteRead(0 bytes read)
May 01 04:30:31 shell mopidy[2722]: ERROR    Uncaught exception
May 01 04:30:31 shell mopidy[2722]: Traceback (most recent call last):
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy/commands.py", line 275, in run
May 01 04:30:31 shell mopidy[2722]: core = self.start_core(mixer, backends, audio)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy/commands.py", line 364, in start_core
May 01 04:30:31 shell mopidy[2722]: return Core.start(mixer=mixer, backends=backends, audio=audio).proxy()
May 01 04:30:31 shell mopidy[2722]: File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 93, in start
May 01 04:30:31 shell mopidy[2722]: obj = cls(*args, **kwargs)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy/core/actor.py", line 52, in __init__
May 01 04:30:31 shell mopidy[2722]: self.backends = Backends(backends)
May 01 04:30:31 shell mopidy[2722]: File "/usr/local/lib/python2.7/dist-packages/mopidy/core/actor.py", line 151, in __init__
May 01 04:30:31 shell mopidy[2722]: has_library = b.has_library().get()
May 01 04:30:31 shell mopidy[2722]: File "/usr/lib/python2.7/dist-packages/pykka/future.py", line 299, in get
May 01 04:30:31 shell mopidy[2722]: exec('raise exc_info[0], exc_info[1], exc_info[2]')
May 01 04:30:31 shell mopidy[2722]: File "<string>", line 1, in <module>
May 01 04:30:31 shell mopidy[2722]: ActorDeadError: GMusicBackend (urn:uuid:2f6b6a7a-32d9-450a-bcbf-e8bd4a18f80b) stopped before handling the message
```

**Comment by lrvick** [comment]
my mopidy.conf:

```
[logging]
config_file = /etc/mopidy/logging.conf
debug_file = /var/log/mopidy/mopidy-debug.log

[local]
enabled = true
data_dir = /var/lib/mopidy/local
media_dir = /var/lib/mopidy/media
playlists_dir = /var/lib/mopidy/playlists

[gmusic]
enabled = true
username = lance@lrvick.net
password = ****************
deviceid = ****************
all_access = true

[mpd]
enabled = true
hostname = 0.0.0.0
port = 6600
password = 
max_connections = 20
connection_timeout = 60
zeroconf = Mopidy MPD server on $hostname
#command_blacklist = listall,listallinfo

[http]
enabled = true
hostname = 0.0.0.0
port = 6680
zeroconf = Mopidy HTTP server on $hostname
```

**Comment by morgoth** [comment]
I have similar error now:

```
2015-05-01 08:40:23,498 INFO [5723:MainThread] mopidy.local.library: Loaded 18057 local tracks using json
2015-05-01 08:40:23,508 INFO [5723:MainThread] mopidy.commands: Starting Mopidy core
2015-05-01 08:40:23,542 INFO [5723:GMusicBackend-3] requests.packages.urllib3.connectionpool: Starting new HTTPS connection (1): www.google.com
2015-05-01 08:40:43,566 ERROR [5723:GMusicBackend-3] pykka: Unhandled exception in GMusicBackend (urn:uuid:703daf7b-1df0-4b31-b49d-826d12cedabf):
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 191, in _actor_loop
    self.on_start()
  File "/usr/local/lib/python2.7/dist-packages/mopidy_gmusic/actor.py", line 48, in on_start
    self.config['gmusic']['deviceid'])
  File "/usr/local/lib/python2.7/dist-packages/mopidy_gmusic/session.py", line 21, in login
    if not self.api.login(username, password):
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/clients/mobileclient.py", line 39, in login
    if not self.session.login(email, password):
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 147, in login
    success = super(Mobileclient, self).login(email, password, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 107, in login
    res = ClientLogin.perform(self, True, email, password)
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/protocol/shared.py", line 208, in perform
    response = session.send(req_kwargs, cls.required_auth)
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 75, in send
    res = self._send_without_auth(req_kwargs, new_session)
  File "/usr/local/lib/python2.7/dist-packages/gmusicapi/session.py", line 42, in _send_without_auth
    return rsession.request(**req_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', gaierror(-2, 'Name or service not known'))
2015-05-01 08:40:43,570 ERROR [5723:MainThread] mopidy.commands: Uncaught exception
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/mopidy/commands.py", line 275, in run
    core = self.start_core(mixer, backends, audio)
  File "/usr/lib/python2.7/dist-packages/mopidy/commands.py", line 364, in start_core
    return Core.start(mixer=mixer, backends=backends, audio=audio).proxy()
  File "/usr/lib/python2.7/dist-packages/pykka/actor.py", line 93, in start
    obj = cls(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/mopidy/core/actor.py", line 52, in __init__
    self.backends = Backends(backends)
  File "/usr/lib/python2.7/dist-packages/mopidy/core/actor.py", line 151, in __init__
    has_library = b.has_library().get()
  File "/usr/lib/python2.7/dist-packages/pykka/future.py", line 299, in get
    exec('raise exc_info[0], exc_info[1], exc_info[2]')
  File "<string>", line 1, in <module>
ActorDeadError: GMusicBackend (urn:uuid:703daf7b-1df0-4b31-b49d-826d12cedabf) stopped before handling the message
2015-05-01 08:40:43,572 INFO [5723:MainThread] mopidy.commands: Stopping Mopidy frontends
2015-05-01 08:40:43,574 INFO [5723:MainThread] mopidy.commands: Stopping Mopidy core
2015-05-01 08:40:43,574 INFO [5723:MainThread] mopidy.commands: Stopping Mopidy backends
2015-05-01 08:40:43,580 INFO [5723:MainThread] mopidy.commands: Stopping Mopidy audio
2015-05-01 08:40:43,584 INFO [5723:MainThread] mopidy.commands: Stopping Mopidy mixer
```

**Comment by Gunbird** [comment]
I also have the same error. With Mopidy_GMusic-0.3.0 from develop and with Mopidy_GMusic-1.0.0 from develop.

I tried with python 2.7.5 and python 2.7.9 as I had warning message with python 2.7.5. I didn't change anything.

mopidy --save-debug-log                                                                                                                                                                                             [26/31]
** Message: pygobject_register_sinkfunc is deprecated (GstObject)
INFO     Starting Mopidy 0.19.5
INFO     Loading config from: builtin defaults, /etc/xdg/mopidy/mopidy.conf, /home/mopidy/.config/mopidy/mopidy.conf, command line options
INFO     Enabled extensions: mpd, http, stream, gmusic, softwaremixer, moped, musicbox_webclient, local
INFO     Disabled extensions: none
INFO     Starting Mopidy mixer: SoftwareMixer
INFO     Mixing using GStreamer software mixing
INFO     Starting Mopidy audio
INFO     Starting Mopidy backends: LocalBackend, StreamBackend, GMusicBackend
INFO     Loaded 11 local playlists from /home/mopidy/Music/Playlists
INFO     Audio output set to "autoaudiosink"
INFO     Loaded 12 local tracks using json
INFO     Mopidy uses Google Music
INFO     Starting Mopidy core
ERROR    Unhandled exception in GMusicBackend (urn:uuid:a7f8e908-6342-47f2-b6fc-ff19fe3e3872):
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/pykka/actor.py", line 191, in _actor_loop
    self.on_start()
  File "/home/mopidy/.local/lib64/python2.7/site-packages/mopidy_gmusic/actor.py", line 48, in on_start
    self.config['gmusic']['deviceid'])
  File "/home/mopidy/.local/lib64/python2.7/site-packages/mopidy_gmusic/session.py", line 21, in login
    if not self.api.login(username, password):
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/clients/mobileclient.py", line 39, in login
    if not self.session.login(email, password):
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/session.py", line 147, in login
    success = super(Mobileclient, self).login(email, password, *args, **kwargs)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/session.py", line 122, in login
    webclient.Init.perform(self, True)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/protocol/shared.py", line 208, in perform
    response = session.send(req_kwargs, cls.required_auth)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/session.py", line 86, in send
    res = self._send_with_auth(req_kwargs, desired_auth, rsession)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/gmusicapi/session.py", line 142, in _send_with_auth
    return rsession.request(**req_kwargs)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/sessions.py", line 605, in send
    r.content
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/models.py", line 750, in content
    self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes()
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/models.py", line 673, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/packages/urllib3/response.py", line 304, in stream
    for line in self.read_chunked(amt):
  File "/home/mopidy/.local/lib64/python2.7/site-packages/requests/packages/urllib3/response.py", line 411, in read_chunked
    raise httplib.IncompleteRead(''.join(line))
IncompleteRead: IncompleteRead(0 bytes read)
ERROR    Uncaught exception
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/mopidy/commands.py", line 273, in run
    core = self.start_core(mixer, backends)
  File "/usr/lib64/python2.7/site-packages/mopidy/commands.py", line 355, in start_core
    return Core.start(mixer=mixer, backends=backends).proxy()
  File "/usr/lib64/python2.7/site-packages/pykka/actor.py", line 93, in start
    obj = cls(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/mopidy/core/actor.py", line 41, in __init__
    self.backends = Backends(backends)
  File "/usr/lib64/python2.7/site-packages/mopidy/core/actor.py", line 112, in __init__
    has_library = b.has_library().get()
  File "/usr/lib64/python2.7/site-packages/pykka/future.py", line 299, in get
    exec('raise exc_info[0], exc_info[1], exc_info[2]')
  File "<string>", line 1, in <module>
ActorDeadError: GMusicBackend (urn:uuid:a7f8e908-6342-47f2-b6fc-ff19fe3e3872) stopped before handling the message
INFO     Stopping Mopidy frontends
INFO     Stopping Mopidy core
INFO     Stopping Mopidy backends
INFO     Stopping Mopidy audio
INFO     Stopping Mopidy mixer

**Comment by Gunbird** [comment]
This seems to be related to this https://github.com/simon-weber/Unofficial-Google-Music-API/issues/323

**Comment by morgoth** [comment]
My problem was due to lack of internet connection. Error message could be more clear though.

**Comment by Gunbird** [comment]
The nproblem is still present and has nothing to do with internet connection here.

Any update on this issue.

**Comment by abarreir** [comment]
@Gunbird Are you still experiencing this issue with the latest develop version?

**Comment by jodal** [comment]
This should have been fixed with the upgrade to gmusicapi 6.0.0 in 0b416d0d5eb9d22864ca0cf8508756b9dd74c9f2.
